tests/workflow_test.go

```diff
@@ -127,6 +129,288 @@ func (s *WorkflowTestSuite) TestStartWorkflowExecution() {
 	})
 }
 
+func (s *WorkflowTestSuite) TestStartWorkflowExecution_UseExisting() {
+	id := "functional-start-workflow-use-existing-test"
+	wt := &commonpb.WorkflowType{Name: "functional-start-workflow-use-existing-test-type"}
+	tq := &taskqueuepb.TaskQueue{
+		Name: "functional-start-workflow-use-existing-test-taskqueue",
+		Kind: enumspb.TASK_QUEUE_KIND_NORMAL,
+	}
+
+	request := &workflowservice.StartWorkflowExecutionRequest{
+		RequestId:          uuid.New(),
+		Namespace:          s.Namespace().String(),
+		WorkflowId:         id,
+		WorkflowType:       wt,
+		TaskQueue:          tq,
+		Input:              nil,
+		WorkflowRunTimeout: durationpb.New(100 * time.Second),
+		Identity:           "worker1",
+	}
+
+	we0, err0 := s.FrontendClient().StartWorkflowExecution(testcore.NewContext(), request)
+	s.NoError(err0)
+	s.True(we0.Started)
+
+	request.RequestId = uuid.New()
+	request.WorkflowIdConflictPolicy = enumspb.WORKFLOW_ID_CONFLICT_POLICY_USE_EXISTING
+	we1, err1 := s.FrontendClient().StartWorkflowExecution(testcore.NewContext(), request)
+	s.NoError(err1)
+	s.Equal(we0.RunId, we1.RunId)
+	s.False(we1.Started)
+}
+
+func (s *WorkflowTestSuite) TestStartWorkflowExecution_UseExisting_OnConflictOptions() {
+	s.OverrideDynamicConfig(
+		callbacks.AllowedAddresses,
+		[]any{map[string]any{"Pattern": "some-secure-address", "AllowInsecure": false}},
+	)
+	cb := &commonpb.Callback{
+		Variant: &commonpb.Callback_Nexus_{
+			Nexus: &commonpb.Callback_Nexus{
+				Url: "https://some-secure-address",
+			},
+		},
+	}
+
+	testCases := []struct {
+		name                     string
+		WorkflowIdConflictPolicy enumspb.WorkflowIdConflictPolicy
+		OnConflictOptions        *workflowpb.OnConflictOptions
+		ErrMessage               string
+		MaxCallbacksPerWorkflow  int
+	}{
+		{
+			name:                     "OnConflictOptions nil",
```

Comment:
Isn't this a duplicate of the test above?

Reply:
Oh, yes. I'll dedup it.